ui/guest/src/react/GuestProxy.tsx

```diff
@@ -196,23 +199,22 @@ export function GuestProxy() {
         document.querySelectorAll<HTMLElement>('[data-craftercms-model-id]').forEach(registerElement);
       });
 
-    const handler: JQuery.EventHandlerBase<any, any> = (e: Event): void => {
-      let record = ElementRegistry.fromElement(e.currentTarget as Element);
+    const handler: JQuery.EventHandlerBase<any, any> = (e: Event, target): void => {
+      let record = ElementRegistry.fromElement(target as Element);
       if (notNullOrUndefined(record)) {
         persistenceRef.current.onEvent(e, record.id);
       }
     };
 
-    $(document)
-      .on('mouseover', '[data-craftercms-model-id]', handler)
-      .on('mouseleave', '[data-craftercms-model-id]', handler)
-      .on('dragstart', '[data-craftercms-model-id]', handler)
-      .on('dragover', '[data-craftercms-model-id]', handler)
-      .on('dragleave', '[data-craftercms-model-id]', handler)
-      .on('drop', '[data-craftercms-model-id]', handler)
-      .on('dragend', '[data-craftercms-model-id]', handler)
-      .on('click', '[data-craftercms-model-id]', handler)
-      .on('dblclick', '[data-craftercms-model-id]', handler);
+    delegateEventListener(document, 'mouseover', '[data-craftercms-model-id]', handler);
+    delegateEventListener(document, 'mouseout', '[data-craftercms-model-id]', handler);
+    delegateEventListener(document, 'dragstart', '[data-craftercms-model-id]', handler);
+    delegateEventListener(document, 'dragover', '[data-craftercms-model-id]', handler);
+    delegateEventListener(document, 'dragleave', '[data-craftercms-model-id]', handler);
+    delegateEventListener(document, 'drop', '[data-craftercms-model-id]', handler);
+    delegateEventListener(document, 'dragend', '[data-craftercms-model-id]', handler);
+    delegateEventListener(document, 'click', '[data-craftercms-model-id]', handler);
+    delegateEventListener(document, 'dblclick', '[data-craftercms-model-id]', handler);
```

Comment:
All these delegates require the most intensive testing. Need to test over and over in all bps and from different angles.